src/Factory.php

```diff
@@ -344,6 +379,32 @@ private function inverseRelationshipField(self $factory): ?string
         return null; // no relationship found
     }
 
+    private function hasCascadePersist(self $factory, ?string $field): bool
+    {
+        if (null === $field) {
+            return false;
+        }
+
+        $collectionClass = $factory->class;
+        $factoryClass = $this->class;
+        $collectionMetadata = self::configuration()->objectManagerFor($collectionClass)->getClassMetadata($collectionClass);
+        $classMetadataFactory = self::configuration()->objectManagerFor($factoryClass)->getMetadataFactory()->getMetadataFor($factoryClass);
+
+        if (false === $collectionMetadata->hasAssociation($field)) {
+            return false;
+        }
+
```

Comment:
I think the below psalm errors can be solved with:

```php
if (!$classMetadata instanceof \Doctrine\ORM\Mapping\ClassMetadataInfo) {
    return false;
}
```

Reply:
Thanks, I've start by use method instead of public attributes to avoid 3 Psalm errors, but this one appear after, and I've not check it.